plugin/httptrace/clienttrace.go

```diff
@@ -77,7 +82,8 @@ func (ct *clientTracer) close(name string) {
 		s.End()
 		delete(ct.levels, name)
 	} else {
-		panic(fmt.Sprintf("failed to find span %s in levels.", name))
+		// open is not finished before close is called.
+		ct.levels[name] = s
```

Comment:
I don't get this assignment, what is `s` here? A `(type: nil, value: nil)` interface? A `(type: trace.Span, value: nil)` interface? Whatever it is, it feels like we are putting a bogus value in the map, and I can't see that we check anywhere if the value we got from the map is valid.

Reply:
Agree with @krnowak, I think we need some test case to validate this.

Reply:
yes. I am going to close this.

Reply:
I have added test for race condition but race condition is not happening even without the fix.

Reply:
> I have added a test for race condition but race condition is not happening even without the fix.

The only way I could repro this issue was by reusing the same http.Request:
```go
req, _ := http.NewRequest("GET", ts.URL, nil)
_, req = httptrace.W3C(ctx, req)
for i := 0; i < 100; i++ {
	res, err := client.Do(req)
	if err != nil {
		return err
	}
	res.Body.Close()
}
```

@lizthegrey are there any chances you're doing this?

Reply:
Now that you mention it, yes, yes my code was doing that.

Reply:
I figured it out. See comments on PR.